### PR TITLE
Exiting if upload fails or is cancelled

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @MDunitz 

**Readability:** @mckinsel 

---

## Changes
- Add exceptions to catch when the processing container fails or processing is canceled.
  - Return 0 if  upload was canceled this will end the upload.
  - Return 1 if upload failed, this will restart the processing container.

### CODE REVIEW

File: backend/corpora/dataset_processing/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class ProcessingCancelled(Exception):
+    def __init__(self, status, *args, **kwargs) -> None:
```
Comment (mckinsel) [inline backend/corpora/dataset_processing/exceptions.py:2]: Can't you just say `pass` here?

File: backend/corpora/dataset_processing/process.py [hunk 2]
```diff
@@ -373,12 +376,20 @@ def process_cxg(local_filename, dataset_id, cellxgene_bucket):
...
+        local_filename = download_from_dropbox_url(
+            dataset_id,
+            os.environ["DROPBOX_URL"],
+            "local.h5ad",
+        )
+    except ProcessingCancelled as ex:
+        logging.info(ex.status)
+        sys.exit(0)
+    except ProcessingFailed as ex:
+        logging.error(ex.status)
```
Comment (mckinsel) [inline backend/corpora/dataset_processing/process.py:389]: What does this do with the actual error and traceback?

Comment (Bento007) [reply backend/corpora/dataset_processing/process.py:389]: the traceback and error are in ex.status.upload_message which are populated by the download code.

**Review by MDunitz — APPROVED** [review]
lgtm!

**Review by mckinsel — APPROVED** [review]
One tiny comment, but not really a big deal either way.

**Pull request merged**